src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,162 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public void unlink(Node<T> node) {
+        if (node == head) {
+            head = node.next;
+            size--;
+        } else if (node == tail) {
+            tail = node.prev;
+            size--;
+        } else {
+            node.next.prev = node.prev;
+            node.prev.next = node.next;
+            size--;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (size == 0) {
+            head = tail = node;
+        } else {
+            node = new Node<>(tail, value, null);
+            tail.next = node;
+            node.prev = tail;
+            node.next = null;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index, size + 1);
+        Node<T> node = new Node<>(null, value, null);
+        Node<T> firstNode = head;
+        if (size == 0) {
+            head = tail = node;
+        } else if (index == 0) {
+            node = new Node<>(null, value, head);
+            node.next = head;
+            head = node;
+        } else if (index == size) {
+            node = new Node<>(tail, value, null);
+            tail.next = node;
+            tail = node;
+        } else {
+            for (int i = 0; i < index - 1; i++) {
+                firstNode = firstNode.next;
+            }
+            node.next = firstNode.next;
+            node.prev = firstNode;
+            firstNode.next.prev = node;
+            firstNode.next = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
+        Objects.checkIndex(index, size);
+        Node<T> countNode = head;
+        int count = 0;
+        while (countNode != null) {
+            if (count == index) {
+                return countNode.value;
+            }
+            count++;
+            countNode = countNode.next;
+        }
         return null;
```

Comment:
Put the logic for this method into a private method that will return a node by index and then the method will look like this
```suggestion
        return getNodeByIndex(index).value;
```